ProjFS.Mac/PrjFSKext/PrjFSKext/PrjFSProviderUserClient.cpp

```diff
@@ -80,6 +80,16 @@ bool PrjFSProviderUserClient::initWithTask(
     return false;
 }
 
+void PrjFSProviderUserClient::stop(IOService* provider)
+{
+    // On the code path of process dying or disconnecting, the following is unnecessary,
+    // but if the kext is unloaded with providers attached, clientClose() will never be
+    // called on them so we do the cleanup here.
```

Comment:
I really appreciate great comments like this.  Makes it very clear whats going on!